executor/batch_point_get.go

```diff
@@ -77,26 +79,38 @@ func (e *BatchPointGetExec) Next(ctx context.Context, req *chunk.Chunk) error {
 }
 
 func (e *BatchPointGetExec) initialize(ctx context.Context) error {
-	var err error
-	// Select for update is not optimized to BatchPointGetExec, so we can use e.startTS here.
-	e.snapshot, err = e.ctx.GetStore().GetSnapshot(kv.Version{Ver: e.startTS})
-	if err != nil {
-		return err
+	e.snapshotTS = e.startTS
+	if e.lock {
+		e.snapshotTS = e.ctx.GetSessionVars().TxnCtx.GetForUpdateTS()
 	}
-
 	txn, err := e.ctx.Txn(false)
 	if err != nil {
 		return err
 	}
-	var reader kv.Snapshot
-	if !txn.Valid() {
+	var batchGetter kv.BatchGetter
+	if txn.Valid() {
 		// In restricted SQL, !txn.Valid() may happen, we doesn't call PrepareTxnCtx but
 		// call Next() and the code runs here.
-		reader = e.snapshot
+		if e.snapshotTS == e.startTS {
```

Comment:
For read-consistency mode,  `BatchPointGetExec.startTS` may be the same with `forUpdateTS`,  here using txn as batchGetter will read wrong data?

Reply:
RC would update the snapshot, so this doesn't affect correctness, but the logic here is useless even for REPEATABLE-READ level, the `e.startTS` is not the real startTS of the transaction we should rename it in the future PR.